tests/base.py

```diff
@@ -33,13 +36,25 @@ def get_type():
     def tap_name():
         return "tap-adroll"
 
-    @staticmethod
-    def get_properties():
-        return {
-            # Start date for ad_reports dating back to 2016
-            'start_date' : '2016-06-03T00:00:00Z',
-            'end_date' : '2016-06-06T00:00:00Z'
+    # @staticmethod
+    # def get_properties():
+    #     return {
+    #         # Start date for ad_reports dating back to 2016
+    #         'start_date' : '2016-06-03T00:00:00Z',
+    #         'end_date' : '2016-06-06T00:00:00Z'
+    def get_properties(self, original: bool = True):
+        return_value = {
+            'start_date' : dt.strftime(dt.utcnow() - timedelta(days=2), self.START_DATE_FORMAT),  # set to utc today
         }
+        if original:
+            return return_value
+
+        # Start Date test needs the new connections start date to be prior to the default
+        assert self.START_DATE < return_value["start_date"]
```

Comment:
I don't see where self.START_DATE is defined. Why go with STATE rather than just pass it in as a parameter. Is one way better than the other?

Reply:
I think we should zoom on this, but the self.START_DATE is defined in the test. The assertion is there to ensure the validity of the start date test. We may be able to pass it in but that is not obvious to me.